packages/webapp/src/components/Animals/AddAnimalsFormCard/AddAnimalsFormCard.tsx

```diff
@@ -78,45 +106,63 @@ export default function AddAnimalsFormCard({
         {showRemoveButton && <SmallButton variant="remove" onClick={onRemoveButtonClick} />}
       </div>
       <AnimalTypeSelect
-        name={`${namePrefix}.${FIELD_NAMES.TYPE}`}
+        name={`${namePrefix}.${BasicsFields.TYPE}`}
         control={control}
         typeOptions={typeOptions}
-        onTypeChange={onTypeChange}
+        onTypeChange={(option) => {
+          trigger(`${namePrefix}.${BasicsFields.TYPE}`);
```

Comment:
I think so, but I only added it to fix the timing of the `Required` error (so that it appears as soon as the input is cleared and resolves when a type is selected) so I'm not sure the identity of the type has been relevant yet... is there a reason you ask? Do you think it would be better in the Controller `field.onChange`?

Reply:
I commented without looking into `AnimalTypeSelect`, but I understand it now!
Initially, I wondered if the trigger should be called after `onTypeChange`, since I thought `onTypeChange` was responsible for setting the value (even though it shouldn't be, since it's optional). It's all good now!!